src/App.jsx

```diff
@@ -45,13 +57,44 @@ function App() {
 		}
 	}, []);
 
-	useEffect(() => {
-		loadCourses();
-		loadAuthors();
+	const loadUser = useCallback(async () => {
+		const token = localStorage.getItem('token');
+		if (token) dispatch(setUser());
 	}, []);
 
-	const addCourseHandler = (newCourse) => {
-		dispatch(addCourseAction(newCourse));
+	useEffect(() => {
+		async function fetchData() {
+			await loadCourses();
+			await loadAuthors();
+			await loadUser();
+		}
+		fetchData();
+	}, [dispatch]);
+
+	const addCourseHandler = async (newCourse) => {
+		if (token) {
+			try {
+				const {
+					data: { result },
+				} = await addCourse(newCourse, token);
+				dispatch(addCourseAction(result));
+			} catch (e) {
+				console.error('failed to add a course', e, newCourse);
+			}
+		}
+	};
+
+	const updateCourseHandler = async (courseId, course) => {
+		if (token) {
+			try {
+				const {
+					data: { result },
+				} = await updateCourse(courseId, course, token);
+				dispatch(updateCourseAction(result));
+			} catch (e) {
+				console.error('failed to update a course', e, course);
+			}
+		}
 	};
```

Comment:
These 2 callbacks belong to the `CourseForm`.
Also, it would be better to create `UpdateCourseForm` and `CreateCourseForm` that would reuse `CourseForm` and provide texts and callbacks for their use-cases.